aten/src/ATen/cpu/vec256/vec256_int.h

```diff
@@ -90,15 +146,76 @@ struct Vec256<int16_t> : public Vec256i {
     vec.load(ptr);
     return vec;
   }
-  void load_partial(const void *ptr, int count) {
-    int16_t tmp_values[size];
+  template <int64_t count>
+  void load(const void* ptr) {
+    __at_align32__ int16_t tmp_values[size];
     std::memcpy(tmp_values, ptr, count * sizeof(int16_t));
-    load(tmp_values);
+    for (int64_t i = count; i < size; i++)
+      tmp_values[i] = _mm256_extract_epi16(values, i);
+    values = _mm256_load_si256(reinterpret_cast<const __m256i*>(tmp_values));
   }
-  void store_partial(void* ptr, int count) const {
-    __at_align32__ int16_t tmp_values[size];
-    store(tmp_values);
-    std::memcpy(ptr, tmp_values, count * sizeof(int16_t));
+  void load(const void* ptr, int64_t count = size) {
+    // TODO: Add bounds checking (error on switch default)?
+    switch (count) {
+      case 0:
+        break;
+      case 1:
+        load<1>(ptr);
+        break;
+      case 2:
+        load<2>(ptr);
+        break;
+      case 3:
+        load<3>(ptr);
+        break;
+      case 4:
+        load<4>(ptr);
+        break;
+      case 5:
+        load<5>(ptr);
+        break;
+      case 6:
+        load<6>(ptr);
+        break;
+      case 7:
+        load<7>(ptr);
+        break;
+      case 8:
+        load<8>(ptr);
+        break;
+      case 9:
+        load<9>(ptr);
+        break;
+      case 10:
+        load<10>(ptr);
+        break;
+      case 11:
+        load<11>(ptr);
+        break;
+      case 12:
+        load<12>(ptr);
+        break;
+      case 13:
+        load<13>(ptr);
+        break;
+      case 14:
+        load<14>(ptr);
+        break;
+      case 15:
+        load<15>(ptr);
+        break;
+      case 16:
+        values = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(ptr));
```

Comment:
Does this really help with anything? I have no idea why do we need it. It seems for me like it can only save us a single shift + sub, but those are essentially free.